### ISSUE TITLE
404 error from NPM Install.

### ISSUE THREAD

**Issue by hydeenoble** [sktan/aws-codeartifact-proxy#4]
Hello, 

I tried to use this proxy (running localhost for testing purposes) to install a scoped package from codeartifact and I keep getting the error message below:

```
'@my-org/<package-name>@http://localhost:8080/@my-org/<package-name>/-/<package-name>-1.26.1.tgz' is not in this registry.
```

The NPM logs shows a `404` error and my `.npmrc` file looks like this: 
```
registry=https://registry.npmjs.org
@my-org:registry=http://localhost:8080/
```

Is there something I am missing?

**Comment by sktan** [comment]
Hey @hydeenoble,

Are you able to download your scoped package from code artifact directly?

If so, are you able to provide some log output from the proxy service, and see if there are any differences in the URLs that are generated when downloading directly vs what the proxy is resolving.

**Comment by aureq** [comment]
A likely explanation to this issue is that AWS CodeArtifact redirects to an S3 bucket URL (presigned url) when downloading the package archive (.tgz) but the proxy changes that URL with it's own. The rewrite should only work *if* the redirect `location` matches the CodeArtifact hostname.

I believe this is the block to look at https://github.com/sktan/aws-codeartifact-proxy/blob/master/src/tools/proxy.go#L61-L69

**Comment by sktan** [comment]
@AkkermanD and @hydeenoble I've released a v0.3.0 version of the proxy which I'm hoping addresses your issue.
